### PR TITLE
gtklock: Fix screen goes black after wrong passwords

### PR DESCRIPTION
**Pull request by vunnyso**
## Description of changes
This patch will fix SP-4849 issue.

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [x] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [x] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status

## Testing

Testing instructions in SP-4849

### CODE REVIEW

**Comment by vunnyso** [comment]
As discussed with @vadika, he suggested to merge the partial fix as of now. We can update with proper fix once its available in https://github.com/jovanlanik/gtklock/issues/101

**Comment by vunnyso** [comment]
> Code is OK.
> 
> Tested on X1: The messages are in a box and doesn't overflow. But screen still goes black after many attempts, but the limit seems to be higher.

Thanks @humaidq-tii for testing, appreciate your help. Yes if we try too many attempts there was race case possibility before which was resulting in coredump and was [fixed](https://github.com/jovanlanik/gtklock/commit/d22127f0fd61bbeba8c12378b3c5b46cc3064d63) in gtklock repo but was not included in release. Now I have included that [patch](https://github.com/tiiuae/ghaf/pull/694/files#diff-493e191c2c7b7accf4f429f738cb243599ac50ebc436bab562c242b89ea3eb01R1-R36) as well. I don't see any black screen on my side (Tested on X1).

File: overlays/custom-packages/gtklock/default.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# This overlay will apply gtklock patches which are merged in master branch
+# https://github.com/jovanlanik/gtklock/commit/d22127f0fd61bbeba8c12378b3c5b46cc3064d63
+# https://github.com/jovanlanik/gtklock/commit/e0e7f6d5ae7667fcc3479b6732046c67275b2f2f
+# TODO: Remove patches, once there new release for gtlk-lock
```
Comment (brianmcgillion) [inline overlays/custom-packages/gtklock/default.nix:7]: please add a link to track these patches, here https://github.com/tiiuae/ghaf/blob/main/overlays/README.md so we can review when to remove them

Comment (vunnyso) [reply overlays/custom-packages/gtklock/default.nix:7]: Thanks @brianmcgillion, Updated with patches.

**Comment by johannarautanen** [comment]
**lenovo x1, booted ghaf from SSD memory card:**

No more black screen while entering lost of wrong passwords and fingerprints.

There is no scroll bar visible for the failure-texts and cannot "close" the failure-texts, that issues will be handled in https://github.com/jovanlanik/gtklock/issues/101


-automated tests: ok
- all apps launched manually ok

**Pull request merged**